src/nvim/edit.c

```diff
@@ -950,6 +973,15 @@ static int insert_handle_key(InsertState *s)
     break;
 
   case Ctrl_W:        // delete word before the cursor
+    if (bt_prompt(curbuf) && (mod_mask & MOD_MASK_SHIFT) == 0) {
+      // In a prompt window CTRL-W is used for window commands.
+      // Use Shift-CTRL-W to delete a word.
+      stuffcharReadbuff(Ctrl_W);
```

Comment:
Nvim doesn't do this for `:terminal` buffers, it seems rather inconsistent to now do it for prompt buffers.

Reply:
For now, I have removed the commit related to the move by `Ctrl_W` in insert mode from this PR.